File: Java-backend/src/main/java/com/learning/entity/Flashcard.java
package com.learning.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("flashcard")
public class Flashcard {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long noteId;

    private Long userId;

    private String question;

    private String answer;

    private Integer difficulty;

    private Double easeFactor;

    private Integer reviewInterval;

    private Integer reviewCount;

    private LocalDateTime nextReviewAt;

    private LocalDateTime createdAt;
}

File: Vue-frontend/src/types/flashcard.ts
export interface Flashcard {
  id: number
  noteId: number
  userId: number
  question: string
  answer: string
  difficulty: number
  easeFactor: number
  reviewInterval: number
  reviewCount: number
  nextReviewAt: string | null
  createdAt: string
}

export interface FlashcardSaveRequest {
  noteId: number
  cards: {
    question: string
    answer: string
    difficulty: number
  }[]
}
